tiny_dnn/core/kernels/recurrent_cell_op_internal.h

```diff
@@ -0,0 +1,154 @@
+/*
+    Copyright (c) 2013, Taiga Nomi and the respective contributors
+    All rights reserved.
+
+    Use of this source code is governed by a BSD-style license that can be found
+    in the LICENSE file.
+*/
+#pragma once
+
+#include "tiny_dnn/core/params/recurrent_cell_params.h"
+
+namespace tiny_dnn {
+namespace kernels {
+
+inline void recurrent_cell_op_internal(const tensor_t &in_data,
+                                       const tensor_t &prev_h,
+                                       const vec_t &U,
+                                       const vec_t &W,
+                                       const vec_t &V,
+                                       const vec_t &bias,
+                                       const vec_t &c,
+                                       tensor_t &out_data,
+                                       tensor_t &out_h,
+                                       const recurrent_cell_params &params,
+                                       const bool layer_parallelize) {
+  for_i(layer_parallelize, in_data.size(), [&](size_t sample) {
+    const vec_t &in         = in_data[sample];
+    const vec_t &prev_state = prev_h[sample];
+    vec_t &out              = out_data[sample];
+    vec_t &next_state       = out_h[sample];
+
+    for (size_t o = 0; o < params.out_size_; o++) {
+      out[o] = float_t{0};
+
+      // W * h(t-1)
+      for (size_t o_2 = 0; o_2 < params.out_size_; o_2++) {
+        next_state[o] += W[o_2 * params.out_size_ + o] * prev_state[o_2];
+      }
+
+      // U*x(t)
+      for (size_t i = 0; i < params.in_size_; i++) {
+        next_state[o] += U[i * params.out_size_ + o] * in[i];
+      }
+
+      if (params.has_bias_) {
+        next_state[o] += bias[o];
+      }
+    }
+
+    params.activation_->forward_activation(next_state, next_state);
+
+    // V matrix is out_size_ x out_size_
+    for (size_t o = 0; o < params.out_size_; o++) {
```

Comment:
In this loop, you can add a local variable. Accessing to `out[o]` should be minimized.